src/rebase.c

```diff
@@ -988,8 +988,10 @@ static int rebase_commit__create(
 
 		if ((error = rebase->options.signing_cb(&commit_signature, &signature_field,
 				git_buf_cstr(&commit_content), rebase->options.payload)) < 0 &&
-				error != GIT_PASSTHROUGH)
+				error != GIT_PASSTHROUGH) {
+			git_error_set(error, "signing_cb failed");
```

Comment:
The guidelines say that we should provide something in the event the callback didn't, but we should preserve something it might have already set. IIRC there's a helper method for those callback functions that takes care of this dance.

So, 
```c
if (git_error_last() == NULL)
    git_error_set(GIT_ERROR_COMMIT, "commit signing_cb failed");
```

Also, you're conflating error codes (ie. return values) with error classes (ie the `GIT_ERROR_*` values), taken as the 1st parameter of `git_error_set`.

Reply:
Gotcha! Thanks!